### DISCUSSION TITLE
Having Spacy detect alternative names for terms separated by 'and' or 'or'

### DISCUSSION THREAD

**Discussion by haydenso** [explosion/spaCy#11071] [Help: Coding & Implementations]
The dataset I am dealing with is identifying gene terms in a given paragraph of text. In the text, sometimes the author would write about alternative ways the gene can be shown through using conjunctions such as 'and' or 'or. The 'alternative term' do not follow the same structure as the gene terms the model is being trained on (rather different sometimes) so that might be a slight issue. 

What is the best practice for using the model to develop a NER model such that it can identify alternative names in context? How can I use rule-based matching to somehow do this or does it require manually annotations?

Any help is appreciated.

**Comment by polm** [comment]
Sorry for the delayed reply. Can you give an example of the type of structure you see and what you want to match? 

I'm imagining something like "These gene can be XYZ, or ABC", where XYZ is something you are already able to match and ABC is something you haven't seen before. In that case I would recommend using a (rule-based) Matcher pattern to match it and see how much coverage that gets you. It may make it easier to match if you use [`merge_noun_chunks`](https://spacy.io/api/pipeline-functions#merge_noun_chunks) or something first.
